met-api/tests/unit/services/test_submission.py

```diff
@@ -116,3 +118,25 @@ def test_submissions_with_comment_are_not_auto_approved(session):  # pylint:disa
 
     assert submission is not None
     assert submission.comment_status_id == Status.Pending.value
+
+
+def test_check_if_submission_has_comments(session):
+    """Make sure comments in submission are approved."""
+    survey, eng = factory_survey_and_eng_model()
+    email_verification = factory_email_verification(survey.id)
+    user_details = factory_user_model()
+
+    # Create a sample submission with a comment in a text field that starts with 'simpletextarea'
+    submission_request: SubmissionSchema = {
+        'submission_json': {'simplepostalcode': 'abc', 'simpletextfield': 'This is some text',
+                            'simpletextfield2': 'This is some text 2', 'simpletextarea2': 'This is a comment 1',
+                            'simpletextarea1': 'This is a comment 1'},
+        'survey_id': survey.id,
+        'user_id': user_details.id,
+        'verification_token': email_verification.verification_token,
+    }
+
+    submission = SubmissionService().create(email_verification.verification_token, submission_request)
+
+    # Assert that the function returns True since there is a comment in a text field that starts with 'simpletextarea'
+    assert submission is not None
```

Comment:
we could perhaps try fetching the submission and checking that the submission.comments with key simpletextfield2 exist

Reply:
refactored for that